web/client/components/data/identify/enhancers/identify.js

```diff
@@ -70,13 +70,25 @@ const identifyLifecycle = compose(
                 enabled,
                 changeMousePointer = () => {},
                 disableCenterToMarker,
-                onEnableCenterToMarker = () => {}
+                onEnableCenterToMarker = () => {},
+                disabledAlwaysOn,
+                defaultConfiguration,
+                identifyConfigureDefault = () => {}
             } = this.props;
 
             if (enabled) {
                 changeMousePointer('pointer');
             }
 
+            identifyConfigureDefault({
+                enabled: isNil(enabled) && defaultConfiguration?.enabled
+                || enabled
+                || (isNil(defaultConfiguration?.enabled) && isNil(defaultConfiguration?.enabled) && true)
+                || false,
+                ...(!isNil(disabledAlwaysOn) && {disabledAlwaysOn: disabledAlwaysOn}),
+                defaultConfiguration: {...defaultConfiguration}
```

Comment:
This part hard to understand. It is a result of a try and error? 
If so, I think you should refactor it when you understood at all the logic behind.

cfg are automatically passed to plugins as props.

These props are overridden by the `cfg` from "connect" . 
So you have to use one properly to read the configuration from `cfg` to set it the proper "identify.configuration" property, that has the real control of the tool.

To skip this overriding you can;
- use a different name(`defaultConfiguration` i suggested this because maybe it was the easier way, but maybe I confused you more with this
- do it in a wrapper before the connect, that get's the configuration, not yet overridden and trigger the action.
- use connects `mergeProps` to rename it or whatever you prefer.

The important thing is that the action should trigger the default values and get from the prop only the overrides, so the Identify tool is always in a known predictable state (without configurations from pervious mount event, and without having the necessity to specify the whole configuration every time). 

So if this part is confusing you, you can isolate the enhancer outside the connect and do it clean.

Reply:
Hi @offtherailz, i made a small changes, so now, default values only sets if the configs empty or enabled is not persist in redux store. So it didn't update store vie renders and routing (obviuosly). 

- use a different name(defaultConfiguration i suggested this because maybe it was the easier way, but maybe I confused you more with this -- it's already stored with this name in redux, but in many places the data does not go thru the selectors, making it difficult to update.
- do it in a wrapper before the connect, that get's the configuration, not yet overridden and trigger the action. -- It will have same effect and +1 more component wrapped with HOC.
- use connects mergeProps to rename it or whatever you prefer. -- wont make any impact